DDSView/ddsview.cpp

```diff
@@ -519,12 +523,12 @@ HRESULT InitDevice(const TexMetadata& mdata)
     case TEX_DIMENSION_TEXTURE1D:
         if (mdata.arraySize > 1)
         {
-            pshader = g_PS_1DArray;
+            pshader = (BYTE*) g_PS_1DArray;
```

Comment:
Prefer to use C++ casting either ``static_cast`` or ``reinterpret_cast``.

Reply:
I did that typecase because I was getting this error message:

```
:/msys64/home/jpmugaas/directxtex-PKGBUILD/src/directxtex-dec2023/DDSView/ddsview.cpp:522:23: error: invalid conversion from 'const signed char*' to 'const BYTE*' {aka 'const unsigned char*'} [-fpermissive]
  522 |             pshader = g_PS_1DArray;
      |                       ^~~~~~~~~~~~
      |                       |
      |                       const signed char*

```

Reply:
```
pshader = reinterpret_cast<const BYTE*>(g_PS_1DArray);
```